pkg/ccl/backupccl/restore_planning.go

```diff
@@ -802,6 +802,44 @@ func resolveTargetDB(
 	return database.Name, nil
 }
 
+// maybeUpgradeTableDescsInSlice updates the passed slice of table descriptors
+// to use the newer 19.2-style foreign key representation, if they are not
+// already upgraded. This requires resolving cross-table FK references, which is
+// done by looking up all table descriptors in the slice provided.
+//
+// if skipFKsWithNoMatchingTable is set, FKs whose "other" table is missing from
+// the set provided are omitted during the upgrade, instead of causing an error
+// to be returned.
+func maybeUpgradeTableDescsInSlice(
+	ctx context.Context, descs []catalog.Descriptor, skipFKsWithNoMatchingTable bool,
+) error {
+	descGetter := catalog.MapDescGetter{}
+
+	// Populate the protoGetter with all table descriptors in all backup
+	// descriptors so that they can be looked up.
+	for _, desc := range descs {
+		if table := descpb.TableFromDescriptor(desc.DescriptorProto(), hlc.Timestamp{}); table != nil {
+			descGetter[table.ID] = tabledesc.NewImmutable(*protoutil.Clone(table).(*descpb.TableDescriptor))
```

Comment:
@postamar I took another swing at this and it seems to compile, but I'm mostly just grabbing the wrap/unwrap methods blindly here